app/api/server/v1/im.js

```diff
@@ -32,10 +32,18 @@ function findDirectMessageRoom(params, user) {
 
 API.v1.addRoute(['dm.create', 'im.create'], { authRequired: true }, {
 	post() {
-		const findResult = findDirectMessageRoom(this.requestParams(), this.user);
+		const { username, usernames } = this.requestParams();
+
+		const users = username ? [username] : usernames && usernames.split(',').map((username) => username.trim());
+
+		if (!users) {
+			throw new Meteor.Error('error-room-not-found', 'The required "roomId", "username" or "usernames" param provided does not match any direct message');
```

Comment:
Where are you checking for the `roomId`?

Reply:
sorry why do we need a roomID? to create an im?
there is no reference about
https://rocket.chat/docs/developer-guides/rest-api/im/create/

Reply:
I don't know, you are referencing it in the error message 😄

Reply:
> Where are you checking for the roomId? there is no expected param called roomId
https://rocket.chat/docs/developer-guides/rest-api/im/create/

Reply:
I just kept the same message we were sending before. removed